### DISCUSSION TITLE
Elastic Windows Agent not working SO v2.4.10

### DISCUSSION THREAD

**Discussion by plaotec** [Security-Onion-Solutions/securityonion#11267] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

32

### Storage for /

250

### Storage for /nsm

250

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Elastic Windows Agent not working SO v2.4.10

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by defensivedepth** [comment]
Where are you installing Elastic Agent? Have you enabled firewall access?

**Comment by jwmasekre** [comment]
You need way more details. Is the installer failing? Is there a log you can share, or at least an error message? Is it a specific component that isn't working?

**Comment by plaotec** [comment]
![Screenshot 2023-09-06 145321](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/633a67ad-5c5f-4b23-9d93-7ce9121ce589)
![Screenshot 2023-09-06 145248](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/2bf547fb-83e6-467f-b2e6-259105fe8ca3)
![salt-pillar-global](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/b2ff0f3f-cef0-49f4-a7a6-cbe48c87fbe8)

**Reply by jwmasekre** [reply]
Can you see the host in Elastic Fleet?

**Comment by plaotec** [comment]
No i don't thanks 

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/ceb85d50-d36d-4c6b-a4ee-5d1c51ae1c46)

**Reply by jwmasekre** [reply]
Okay, is there an `SO-Elastic-Agent-Installer.log` file in the directory the installer was run from?

**Comment by plaotec** [comment]
timestamp=2023-09-05T11:37:22.7252754-05:00 level=info message="Version Information" ElasticAgentVersion=8.7.0 WrapperVersion=2.4.2
timestamp=2023-09-05T11:37:22.7264295-05:00 level=info message="Runtime Data" EnrollmentToken="YU1WTlFvb0JSejY2Y3dfY0hTLUk6NmQ5TVFGWF9UVW1RcjIycnJ1T1JoUQ==" FleetURL/s=https://198.18.133.230:8220,https://securityonion:8220
timestamp=2023-09-05T11:37:22.7264295-05:00 level=info message="Installation Progress" Status="Starting Installation Precheck"
timestamp=2023-09-05T11:37:25.7438057-05:00 level=warn message="Installation Progress" FleetHostConnectivityCheck=Failed FleetHostURL=https://198.18.133.230:8220
timestamp=2023-09-05T11:37:28.0401855-05:00 level=warn message="Installation Progress" FleetHostConnectivityCheck=Failed FleetHostURL=https://securityonion:8220
timestamp=2023-09-05T11:37:28.0457381-05:00 level=info message="Installation Progress" Status="No Fleet Hosts are accessible - Check connectivity to Fleet Host."
timestamp=2023-09-05T11:37:28.0462486-05:00 level=info message="Installation Progress" Status="Exiting Installer..."

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/b08d4f1d-7e25-4430-9b33-342790d09bec)
 let me check connectivity .. windows host can't reach SO .. thanks!

**Comment by plaotec** [comment]
Do we also have to like disable the Oracle Linux Firewall?

**Reply by jwmasekre** [reply]
If you installed from the ISO, you do not need to manually adjust the firewall on the OS. All firewall configurations should be done in the Adminsitration->Configuration page in the webapp.

**Comment by plaotec** [comment]
I disabled iptable on SO VM, can ping and https to SO now but still not able to https to port 8220

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/f37d0a05-d06f-458d-9b03-caea8d6f7af0)

**Reply by jwmasekre** [reply]
Can you confirm the IP address of the endpoint you're installing the agent on? Also, https to 8220 should give you a 404.

**Reply by plaotec** [reply]
endpoint ip is 198.19.10.21

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/6e6d9471-8e4e-4144-86c0-2db3ee8fa840)

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/bb1f211b-2ccd-4d18-a42f-64a13e42e423)

**Comment by plaotec** [comment]
from 198.19.10.21 I can browse to SO GUI and ping SO IP, but not https to port 8220

**Comment by plaotec** [comment]
Thanks

**Comment by plaotec** [comment]
I stop iptables again .. now i get the 404 error let me try agent install again

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/f733b244-5600-40ec-bb84-b850c4805c9e)

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/e680a2e9-52da-48ce-a196-0d8e12c5c45a)

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/7969a79d-20c2-4571-8267-af0f92a455da)

**Comment by defensivedepth** [comment]
@plaotec stopping iptables or manually managing iptables is not the correct way to fix this. Security Onion manages the iptables state and will re-enabled it when a salt highstate runs (every 15 min).

Can you please post the output of the following: (Run on your Manager)

`sudo cat /opt/so/saltstack/local/pillar/firewall/soc_firewall.sls`

**Comment by plaotec** [comment]


**Reply by plaotec** [reply]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/a32578cc-dedb-42aa-8f9d-0aaf5f9981e4)


and agent showing unhealthy status now

**Reply by jwmasekre** [reply]
https://github.com/Security-Onion-Solutions/securityonion/discussions/11148

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/30a20fab-ce27-47b2-b6f2-35f8ec5cded5)

**Comment by plaotec** [comment]
some how SO firewall not working properly with iptables?

**Comment by plaotec** [comment]
let me try to change the SO firewall hostgroup to permit the whole subnet

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/3280122b-fb54-4f4a-adc7-77899a63369f)

**Comment by plaotec** [comment]
iptables

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/49944d55-5e42-4a6f-a1ee-b23ce38a69e3)

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/49d5d4ef-9c8c-4ede-8ea6-ba45034ad627)

**Comment by plaotec** [comment]
back to unhealthy again 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/30cda6b2-aeaf-4028-96bd-14346a12dc91)

**Comment by plaotec** [comment]
iptables now

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/41783bac-2c8b-4534-9992-20a490d9abed)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/b3b17aa9-62d4-4418-88b6-bc084fe50ff9)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/a2fc8157-c05d-4f6b-9e2e-f01cf3a5c96a)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/54c8e783-e0b6-4c9b-9b6f-788c326a4595)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/dce71665-26d9-4c04-a55e-cb5af8bbb42c)


![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/b1a64eec-7387-43c2-a4f5-fe203ebbdbed)

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/2befdc28-aa42-42ca-9e87-091b234ff31b)

**Reply by plaotec** [reply]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/63f31a55-9542-4a83-acd3-f84c4b66b3ed)



some config missing at the agent's end?

**Reply by plaotec** [reply]
agent running on Windows

**Reply by defensivedepth** [reply]
Have you installed the hotfix?

**Reply by plaotec** [reply]
have not, will try to update Windows .. thanks

**Reply by defensivedepth** [reply]
Hotfix for Security Onion. What specific version of SO are you running?

**Reply by plaotec** [reply]
v2.4.10

**Reply by defensivedepth** [reply]
Please post the output of the following, run on your Manager:

`sudo cat /etc/sohotfix`

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/6888cdee-b6fa-486b-a25e-6ad070f9268d)

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/aeff5b08-526c-4541-ae7e-ce27896929b1)

**Comment by plaotec** [comment]
we are on 2.4.10-20230821

**Comment by plaotec** [comment]
where can i find the hot fix for 2.4.10-20230821 and how to apply it? thanks again

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/08376750-6285-4860-9318-ee85d2aba61b)

**Reply by defensivedepth** [reply, marked as answer]
Run `sudo soup` on the Manager

**Reply by plaotec** [reply]
Do I need to mount the latest SO iso to do the soup? What and where is the latest SO v2.4 ISO?...thanks

**Reply by defensivedepth** [reply]
If you are not setup as Airgap, you can just run `sudo soup` on the Manager.
